pkg/common/cns-lib/volume/manager.go

```diff
@@ -2502,21 +2513,32 @@ func (m *defaultManager) createSnapshotWithImprovedIdempotencyCheck(ctx context.
 		SnapshotID:                          snapshotCreateResult.Snapshot.SnapshotId.Id,
 		SourceVolumeID:                      snapshotCreateResult.Snapshot.VolumeId.Id,
 		SnapshotDescription:                 snapshotCreateResult.Snapshot.Description,
-		AggregatedSnapshotCapacityInMb:      snapshotCreateResult.AggregatedSnapshotCapacityInMb,
 		SnapshotLatestOperationCompleteTime: *createSnapshotsTaskInfo.CompleteTime,
 	}
-
-	if m.idempotencyHandlingEnabled {
-		// create the volumeOperationDetails object for persistence
+	if isStorageQuotaM2FSSEnabled && m.clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
```

Comment:
Open a task, add it in our Post Code Complete list

Reply:
its not clear, for what we need to create separate task?